### PR TITLE
core/capabilities/ccip: bump docs, add TODOs for buggy behavior

### PR DESCRIPTION
**Pull request by makramkd**
Update some stale docs that use the blue/green lingo. Also added some TODOs where bugs were found, and did some renames. Added a doc.go for core/capabilities/ccip explaining code organization in the package.

### CODE REVIEW

File: core/capabilities/ccip/launcher/launcher.go [hunk 1]
```diff
@@ -57,8 +58,12 @@ func New(
 type launcher struct {
 	services.StateMachine
 
-	capabilityID    string
-	p2pID           ragep2ptypes.PeerID
+	// capabilityID is the fully qualified capability registry ID of the CCIP capability.
+	// this is <capability_name>@<capability-semver>, e.g "ccip@1.0.0".
+	capabilityID string
```
Comment (makramkd) [inline core/capabilities/ccip/launcher/launcher.go:63]: Should either be a type or add some sort of validation.

File: core/capabilities/ccip/launcher/launcher.go [hunk 2]
```diff
@@ -314,26 +324,24 @@ func updateDON(
...
 	}, nil
 }
 
-// valid cases:
-// a) len(ocrConfigs) == 2 && !prevDeployment.HasCandidateInstance(pluginType): this is a new candidate instance.
-// b) len(ocrConfigs) == 1 && prevDeployment.HasCandidateInstance(): this is a promotion of candidate->active.
-// All other cases are invalid. This is enforced in the ccip config contract.
+// TODO: this is not technically correct, CCIPHome has other transitions
+// that are not covered here, e.g revokeCandidate.
 func createFutureActiveCandidateDeployment(
```
Comment (makramkd) [inline core/capabilities/ccip/launcher/launcher.go:345]: Bug No. 1

File: core/capabilities/ccip/launcher/launcher.go [hunk 3]
```diff
@@ -231,24 +239,26 @@ func (l *launcher) processAdded(added map[registrysyncer.DonID]registrysyncer.DO
...
+			return fmt.Errorf("processAdded: call createDON %d: %w", donID, err)
 		}
 		if dep == nil {
 			// not a member of this DON.
 			continue
 		}
 
+		// TODO: this doesn't seem to be correct; a newly added DON will not have an active
+		// instance but a candidate instance.
 		if err := dep.StartActive(); err != nil {
```
Comment (makramkd) [inline core/capabilities/ccip/launcher/launcher.go:257]: Bug No. 2

File: core/capabilities/ccip/launcher/launcher.go [hunk 4]
```diff
@@ -409,6 +417,7 @@ func createDON(
 		return nil, fmt.Errorf("failed to create CCIP exec oracle: %w", err)
 	}
 
+	// TODO: incorrect, should be setting candidate?
```
Comment (makramkd) [inline core/capabilities/ccip/launcher/launcher.go:420]: Bug No. 3

File: core/capabilities/ccip/launcher/launcher.go (hunk 2 above)
Comment (makramkd) [reply core/capabilities/ccip/launcher/launcher.go:345]: Added to https://smartcontract-it.atlassian.net/browse/CCIP-2687

File: core/capabilities/ccip/launcher/launcher.go (hunk 3 above)
Comment (makramkd) [reply core/capabilities/ccip/launcher/launcher.go:257]: https://smartcontract-it.atlassian.net/browse/CCIP-2687

File: core/capabilities/ccip/launcher/launcher.go (hunk 4 above)
Comment (makramkd) [reply core/capabilities/ccip/launcher/launcher.go:420]: https://smartcontract-it.atlassian.net/browse/CCIP-2687

**Review by dimkouv — APPROVED** [review]

File: core/capabilities/ccip/launcher/launcher.go [hunk 5]
```diff
@@ -57,8 +58,12 @@ func New(
...
 	services.StateMachine
 
-	capabilityID    string
-	p2pID           ragep2ptypes.PeerID
+	// capabilityID is the fully qualified capability registry ID of the CCIP capability.
+	// this is <capability_name>@<capability-semver>, e.g "ccip@1.0.0".
+	capabilityID string
+
+	// myP2PID is the peer ID of the node running this launcher.
+	myP2PID         ragep2ptypes.PeerID
```
Comment (winder) [inline core/capabilities/ccip/launcher/launcher.go:66]: nit: "local" instead of "my"

**Review by winder — APPROVED** [review]

**Pull request merged**